### PR TITLE
Add config option `skip_physical_aggregate_schema_check `

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?
Closes https://github.com/apache/datafusion/issues/13065

## Rationale for this change

Some plans that used to run in DataFusion 41.0.0 started erroring in DataFusion 42.0.0 due to a new check that was added.

The delta-rs 42.0.0 upgrade has it this https://github.com/delta-io/delta-rs/pull/2886#issuecomment-2425616646, and we hit the same thing in InfluxDB IOx as well.

To help unblock upgrades, I would like to permit users to optionally disable this check. 

This config flag is meant to be temporary while we fix all the underlying bugs. 

Background: a new check for exact schema equality added in https://github.com/apache/datafusion/pull/11989 (released in `DataFusion 42.0.0`). This has found quite a few bugs where the schema doesn't quite match due to nullability or metadata mismatch -- see the list on https://github.com/apache/datafusion/issues/12733. There is at least one more bug @wiedld  and I are tracking down. 

After upgrading to DataFusion 42 some of these plans now error (due to bugs in DataFusion, see list on https://github.com/apache/datafusion/issues/12733)

## What changes are included in this PR?

Add a `skip_physical_aggregate_schema_check` option to disable this check so downstream users can workaround any issues they hit

## Are these changes tested?

The doc is tested by CI. Since this a workaround for bugs

## Are there any user-facing changes?
A new config option

### CODE REVIEW

**Comment by alamb** [comment]
I plan to backport this option to the 42 branch as well defaulted to `true` to unblock the delta-rs upgrade

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

**Comment by alamb** [comment]
See additional context here: https://github.com/apache/datafusion/issues/13065#issuecomment-2445404287

File: datafusion/common/src/config.rs [hunk 1]
```diff
@@ -268,6 +268,17 @@ config_namespace! {
         /// Defaults to the number of CPU cores on the system
         pub planning_concurrency: usize, default = num_cpus::get()
 
+        /// When set to true, skips verifying that the schema produced by
+        /// planning the input of `LogicalPlan::Aggregate` exactly matches the
+        /// schema of the input plan.
+        ///
```
Comment (comphead) [inline datafusion/common/src/config.rs:274]: wondering if blank lines can break the formatting and tests eventually failed

Comment (alamb) [reply datafusion/common/src/config.rs:274]: I think the reason the tests failed before is that I forgot to update the `information_schema.slt` test

**Review by jayzhan211 — APPROVED** [review]

**Comment by alamb** [comment]
Thanks again @comphead  and @jayzhan211

**Comment by alamb** [comment]
PR to backport https://github.com/apache/datafusion/pull/13189

**Comment by findepi** [comment]
Looking at https://github.com/apache/datafusion/issues/13190
It seems that schema check is too strict (It should allow fulfilling nulllable contract with non-null data, but not the other way around).

If we relax the check around this, how many problems would remain to be addressed?

**Comment by alamb** [comment]
> Looking at #13190 It seems that schema check is too strict (It should allow fulfilling nulllable contract with non-null data, but not the other way around).
> 
> If we relax the check around this, how many problems would remain to be addressed?

It seems to me the check is finding real things (where the schema of the `ExecutionPlan` doesn't match the schema of the `LogicalPlan` from when it was created)

However, it also seems to me like maybe the discrepancy isn't a huge deal (so maybe it shouldn't be an error 🤔 )

**Comment by findepi** [comment]
Real things like type mismatch, or real things like stricter non-null guarantees (which may be considered an optimization, not an error)?

**Comment by alamb** [comment]
> Real things like type mismatch, or real things like stricter non-null guarantees (which may be considered an optimization, not an error)?

What I know of was related to non-null guarantees as well as user defined metadata (on the Schema and the Field)

**Pull request merged**